Support quoted Content-Disposition filenames for downloads

jimpeters opened the pull request:
1️⃣ Is this something that is wanted/needed? Did you create an issue / discussion about it first?

Needed.  Yes. #1602 

2️⃣ Does it contain multiple, unrelated changes? Please separate the PRs out.

No.

3️⃣ Does it include tests, if possible? (Not a deal-breaker, just a nice-to-have)

Yes

4️⃣ Please include a thorough description of the improvement and reasons why it's useful.

You are unable to download a file in Livewire using a filename like "This & That.txt".  The contents will download but the filename will be surrounded by quotes in the JSON response to Livewire causing weird filename behavior on the saved computer.  Quotes will be replaced with underscores in the filename, etc.

The Content-Disposition header should contain quotes around the filename when special characters or spaces are involved but it is not required when dealing with simple filenames.  It seems Laravel responds with and without quotes depending on the situation.

Content-Disposition: attachment; filename=filename.jpg
Content-Disposition: attachment; filename="filename.jpg"

Are both valid.

5️⃣ Thanks for contributing! 🙌

jimpeters committed:
Fix Content-Disposition header parsing

This should allow the two valid content-disposition header formats

attachment; filename=test.pdf
attachment; filename="test.pdf"

jimpeters committed:
add unique filename for second test

jimpeters committed:
remove empty line

calebporzio commented on src/RenameMe/SupportFileDownloads.php:
```diff
@@ -20,7 +20,13 @@ function __construct()
             $response = $returned;
 
-            $name = Str::after($response->headers->get('Content-Disposition'), 'filename=');
+            preg_match(
```
Please extract this to a separate function, and maybe add a comment to accompany the regex expression - it's not obvious at first glance

jimpeters replied:
Here is my first attempt.  I also simplified the regexes for more readability since we are in a new method now.

calebporzio replied:
This is much better, thanks!

jimpeters committed:
Extract regex conditions to method and simplify

Original regex was too complicated to easily understand right away.

Broke out into two regex's that are easier to read.

Added a default filename of 'download' in case content-disposition is
trashed for some reason or missing
